iceoryx_examples/complexdata/iox_publisher_vector.cpp

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2021 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "iceoryx_utils/cxx/vector.hpp"
+
+#include "iceoryx_posh/popo/publisher.hpp"
+#include "iceoryx_posh/runtime/posh_runtime.hpp"
+#include "iceoryx_utils/posix_wrapper/signal_handler.hpp"
+
+std::atomic_bool killswitch{false};
+constexpr char APP_NAME[] = "iox-cpp-publisher-vector";
+
+static void sigHandler(int f_sig [[gnu::unused]])
+{
+    // caught SIGINT or SIGTERM, now exit gracefully
+    killswitch = true;
+}
+
+int main()
+{
+    // register sigHandler
+    auto signalIntGuard = iox::posix::registerSignalHandler(iox::posix::Signal::INT, sigHandler);
+    auto signalTermGuard = iox::posix::registerSignalHandler(iox::posix::Signal::TERM, sigHandler);
+
+    // initialize runtime
+    iox::runtime::PoshRuntime::initRuntime(APP_NAME);
+
+    // initialize publisher
+    iox::popo::Publisher<iox::cxx::vector<double, 5>> publisher({"Radar", "FrontRight", "Object"});
```

Comment:
This is still quite near to our examples with a different data type and my inner chicken feels that this might become a problem. Could you please change this even more, like having `VectorData` as event id?